src/sources/kubernetes_logs/pod_metadata_annotator.rs

```diff
@@ -323,6 +348,58 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_annotate_from_pod_status() {
+        let cases = vec![
+            (
+                FieldsSpec::default(),
+                PodStatus::default(),
+                LogEvent::default(),
+            ),
+            (
+                FieldsSpec::default(),
+                PodStatus {
+                    pod_ip: Some("192.168.1.2".to_owned()),
+                    ..Default::default()
+                },
+                {
+                    let mut log = LogEvent::default();
+                    log.insert("kubernetes.pod_ips", "192.168.1.2");
+                    log
+                },
+            ),
+            (
+                FieldsSpec {
+                    pod_node_name: "node_name".to_owned(),
+                    ..Default::default()
+                },
```

Comment:
This test should probably use the `FieldsSpec::default()`, and there should be another one that asserts that custom `pod_ips` (and `pod_ip`) is properly utilized.